modules/pam_access/pam_access.c

```diff
@@ -668,6 +692,40 @@ user_match (pam_handle_t *pamh, char *tok, struct login_info *item)
 }
 
 
+/* group_name_or_gid_match - match a group name or group gid against one token */
+
+static int
+group_name_or_gid_match (pam_handle_t *pamh, const char *tok, const char* usr, int debug)
+{
+    /* check for exact match of group name */
+    if (pam_modutil_user_in_group_nam_nam(pamh, usr, tok) != NO) {
+        return YES;
+    }
+
+
+    if (tok[strspn(tok, "0123456789")] != '\0') {
+        return NO;
+    }
+
+    char *endptr = NULL;
+    errno = 0;
+    unsigned long int grpgid = strtoul(tok, &endptr, 10);
+    if(errno != 0
+    || tok == endptr
+    || grpgid > UINT_MAX
+    ) {
```

Comment:
Something went wrong with the indentation.
Also, `*endptr != '\0'` check is missing.